src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -409,6 +409,19 @@ bool CompilerOracle::should_break_at(const methodHandle& method) {
   return check_predicate(CompileCommand::Break, method);
 }
 
+bool CompilerOracle::should_blackhole(const methodHandle& method) {
+  if (check_predicate(CompileCommand::Blackhole, method)) {
+    if (method->result_type() == T_VOID) {
+      return true;
+    } else {
+      ttyLocker ttyl;
+      tty->print_cr("Warning: blackhole compile command only works for methods with void type: %s",
```

Comment:
It is printed unconditionally. What about guarding it with `!CompilerOracle::be_quiet()`?

Reply:
As far as I can see, `CompilerOracle` prints warnings even with `quiet`. `quiet` seems to really only shun the successful messages. So this behavior is consistent. And frankly, it is a better UX: errors in compiler commands should be visible.

Reply:
IMO `CompileCommand=quiet` is fine here, but another option is to check `PrintWarnings`. I believe the output can be quite noisy and there's no way to turn it off.

Reply:
Well, if users want to shun the warning, then they should not put incorrect compiler commands to JVM. For example, JMH does run with `quiet` to silence the "successful" command output, and I still want JVM to show warning messages when people put a `@CompilerControl(BLACKHOLE)` on a wrong method. It is better UX.

Reply:
Then check `PrintWarnings`? It's an explicit user request to silent JVM warnings on output stream.

Reply:
I agree now that `CompileCommand=quiet` is not appropriate here.

Reply:
All right. I see `CompilerOracle` calls into `warning` from `debug.cpp` that handles `PrintWarnings`. Updated new hunk to use it, and a test case that proves it works.